src/utils/parson_json_utils/src/parson_json_utils.c

```diff
@@ -236,7 +236,7 @@ bool ADUC_JSON_GetUnsignedIntegerField(const JSON_Value* jsonValue, const char*
     val = json_object_dotget_number(jsonObj, jsonFieldName);
 
     // Note: check value is not negative and is a whole number for safe casting to an unsigned int
-    if (val < 0 || ((int)val) != val)
+    if (val <= 0 || ((int)val) != val)
```

Comment:
Zed is a valid non negative whole number. This would prevent any call to get a value of 0 from a json structure.

Reply:
actually, parson returns 0 on failure for the various "get number" APIs, which is why it was not setting to default setting when it was missing in the du-config.json.

That said, I will switch to using json_object_has_value() on these, which is better than using 0